includes/multi-currency/class-multi-currency.php

```diff
@@ -277,11 +248,42 @@ public function get_enabled_currencies() {
 		}
 
 		// Set default currency to the top of the list.
-		$dc                         = $this->get_default_currency();
-		$default[ $dc->get_code() ] = $this->enabled_currencies[ $dc->get_code() ];
-		unset( $this->enabled_currencies[ $dc->get_code() ] );
+		$default[ $default_code ] = $this->enabled_currencies[ $default_code ];
+		unset( $this->enabled_currencies[ $default_code ] );
 		$this->enabled_currencies = array_merge( $default, $this->enabled_currencies );
+	}
 
+	/**
+	 * Sets the default currency.
+	 */
+	private function set_default_currency() {
+		$this->default_currency = $this->available_currencies[ get_woocommerce_currency() ];
```

Comment:
Small nitpick: Perhaps we should add a check here to make sure that `$this->available_currencies[ get_woocommerce_currency() ];` is set before using it to set the default currency? Just to ensure we don't throw an error here if either the `available_currencies` array or the `get_woocommerce_currency()` function doesn't return what we expect it to.

Reply:
Cool, per our discussion updated in https://github.com/Automattic/woocommerce-payments/pull/2006/commits/894bbe62547c80b830324316fbcec54e1bfdd1f0